py-polars/tests/unit/operations/rolling/test_rolling.py

```diff
@@ -220,13 +220,13 @@ def test_rolling_crossing_dst(
 
 def test_rolling_by_invalid() -> None:
     df = pl.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]}).sort("a")
-    msg = "in `rolling_min` operation, `by` argument of dtype `i64` is not supported"
+    msg = r"in `rolling_\*_by` operation, `by` argument of dtype `i64` is not supported"
     with pytest.raises(InvalidOperationError, match=msg):
-        df.select(pl.col("b").rolling_min_by("a", 2))  # type: ignore[arg-type]
+        df.select(pl.col("b").rolling_min_by("a", "2i"))
```

Comment:
now that `rolling_min_by` is split from `rolling_min`, it no longer accepts integers as input for `window_size`, and raises a different error

I've slightly modified this one to keep the spirit of the existing test (i.e. asserting that the `by` column can't be of integer dtype)